kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/DataFileStatisticsSuite.scala

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util
+
+import java.util.Collections
+
+import scala.collection.JavaConverters.mapAsJavaMapConverter
+
+import io.delta.kernel.exceptions.KernelException
+import io.delta.kernel.expressions.{Column, Literal}
+import io.delta.kernel.statistics.DataFileStatistics
+import io.delta.kernel.types._
+
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.must.Matchers
+
+class DataFileStatisticsSuite extends AnyFunSuite with Matchers {
+
+  val objectMapper = new ObjectMapper()
+
+  def jsonToNode(json: String): JsonNode = {
+    objectMapper.readTree(json)
+  }
+
+  def areJsonNodesEqual(json1: String, json2: String): Boolean = {
+    val node1 = jsonToNode(json1)
+    val node2 = jsonToNode(json2)
+    node1 == node2
+  }
+
+  test("DataFileStatistics serialization with all types") {
+    val nestedStructType = new StructType()
+      .add("aa", StringType.STRING)
+      .add("ac", new StructType().add("aca", IntegerType.INTEGER))
+
+    val schema = new StructType()
+      .add("ByteType", ByteType.BYTE)
+      .add("ShortType", ShortType.SHORT)
+      .add("IntegerType", IntegerType.INTEGER)
+      .add("LongType", LongType.LONG)
+      .add("FloatType", FloatType.FLOAT)
+      .add("DoubleType", DoubleType.DOUBLE)
+      .add("DecimalType", new DecimalType(10, 2))
+      .add("StringType", StringType.STRING)
+      .add("DateType", DateType.DATE)
+      .add("TimestampType", TimestampType.TIMESTAMP)
+      .add("TimestampNTZType", TimestampNTZType.TIMESTAMP_NTZ)
+      .add("BinaryType", BinaryType.BINARY)
+      .add("NestedStruct", nestedStructType)
+
+    val minValues = Map(
+      new Column("ByteType") -> Literal.ofByte(1.toByte),
+      new Column("ShortType") -> Literal.ofShort(1.toShort),
+      new Column("IntegerType") -> Literal.ofInt(1),
+      new Column("LongType") -> Literal.ofLong(1L),
+      new Column("FloatType") -> Literal.ofFloat(0.1f),
+      new Column("DoubleType") -> Literal.ofDouble(0.1),
```

Comment:
please add all variants of the double: NaN, infinity. Same for float.

Reply:
I think when I added `DefaultJsonHandlerSuite` I was pretty thorough about edge cases for various types if it helps the tests are here https://github.com/delta-io/delta/blob/master/kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/engine/DefaultJsonHandlerSuite.scala